src/client/errors.ts

```diff
@@ -3,79 +3,98 @@
+     * was reached waiting for it to handle the provided context.
      */
-    AppTimeout = 'AppTimeout',
+    SendContextTimeout = 'SendContextTimeout',
+
     /**
-     * Currently unused.
+     * In the case where the optional `context` argument is provided, indicates that the provided application was started, but did not add
+     * a context handler.
      */
-    ResolverUnavailable = 'ResolverUnavailable',
+    SendContextNoHandler = 'SendContextNoHandler'
 }
 
 /**
- * Error codes returned after failure to fire intents to applications.
+ * Error codes specific to the [[raiseIntent]] function.
  */
-export enum ResolveError {
+export enum RaiseIntentError {
     /**
-     * Indicates that no apps could be found for a particular intent.
+     * Indicates that no application could be found to handle the provided intent and context.
      */
     NoAppsFound = 'NoAppsFound',
     /**
-     * Indicates that, in the case when a 'target' argument is passed to [[raiseIntent]], no
-     * such app either exists in the application directory or is currently running.
+     * In the case when a the optional 'target' argument is provided, no such application either exists in the application directory or is
+     * currently running.
      */
     TargetAppNotAvailable = 'TargetAppNotAvailable',
     /**
-     * Indicates that, in the case when a 'target' argument is passed to [[raiseIntent]], the app is not able to handle this intent.
+     * In the case when a the optional 'target' argument is provided, indicates that the app is not able to handle this intent and context.
      */
     TargetAppDoesNotHandleIntent = 'TargetAppDoesNotHandleIntent',
     /**
-     * Indicates that the provider has started an app to receive an intent, and that it has timed out whilst waiting
-     * for that app to handle the intent.
+     * The intent resolver UI was dismissed by the user, so the intent has been cancelled.
      */
-    IntentTimeout = 'IntentTimeout',
+    ResolverClosedOrCancelled = 'ResolverClosedOrCancelled',
     /**
-     * Currently unused.
+     * Indicates that an application was started, but threw an error when attempting to handle the provided intent and context.
      */
-    ResolverUnavailable = 'ResolverUnavailable',
+    SendIntentError = 'SendIntentError',
     /**
-     * Currently unused.
+     * Indicates that an application was started, but a timeout was reached waiting for it to handle the provided intent and context.
      */
-    ResolverTimeout = 'ResolverTimeout',
+    SendIntentTimeout = 'SendIntentTimeout',
     /**
-     * The intent resolver UI was dismissed by the user, so the intent has been cancelled.
+     * Indicates that an application was started, but did not add a handler for the provided intent.
      */
-    ResolverClosedOrCancelled = 'ResolverClosedOrCancelled',
+    SendIntentNoHandler = 'SendIntentNoHandler'
+}
+
+/**
+ * Error codes relating to launching applications.
```

Comment:
Again, possibly one for another story, but bugs me that we're mixing our approaches here. Seems weird that some error enums are API-specific (but coverage is far smaller than the API as a whole), and some error enums are "generic".

For `ApplicationError` specifically, for example - it's not clear if `open` or `raiseIntent` can throw `ApplicationError`s. Both those APIs can start applications, but these errors don't exist in the API-specific enums above. Or, possibly worse, they have duplicate error codes for the same error - making it hard to handle errors in a somewhat generic way.

Reply:
Some of the errors very much are API call specific and make sense to present that, but if we were to go along with that with full consistency, then we're getting into making call-specific TypeErrors, for instance

Reply:
Not completely sure I agree - but yeah, the answer could just be some clarifying comments. Can decide as part of SERVICE-561.